### PR TITLE
Add string.prototype.at polyfill (stage3)

### PR DESCRIPTION
**Pull request by caub**

### CODE REVIEW

File: src/plugins/js-eval/Dockerfile [hunk 1]
```diff
@@ -12,7 +12,7 @@ RUN node_version=${node_version} \
 RUN node --version
 
 # install shims, and acorn+nearley needed by engine262
-RUN npm i --prefix /run airbnb-js-shims@latest acorn@latest nearley@latest
+RUN npm i --prefix /run airbnb-js-shims@latest string.prototype.at@latest acorn@latest nearley@latest
```
Comment (ljharb) [inline src/plugins/js-eval/Dockerfile:15]:
```suggestion
RUN npm i --prefix /run airbnb-js-shims@latest string.prototype.at@latest array.prototype.at@latest acorn@latest nearley@latest
```

**Review by ljharb — COMMENTED** [review]
why would installing it be sufficient to shim it tho?

**Comment by caub** [comment]
it's not sufficient indeed, I'll add `node -r string.prototype.at -r array.prototype.at` maybe

**Comment by ljharb** [comment]
There’s lots of other things to polyfill - but i don’t think that should be done in n> - it should really only be in b>

File: src/plugins/js-eval/jsEvalPlugin.js [hunk 2]
```diff
@@ -64,6 +64,7 @@ module.exports = async function jsEvalPlugin({ mentionUser, respond, message })
       '--experimental-vm-modules', // used by m>
       '--experimental-modules',
       '--no-warnings',
+      '-r', 'string.prototype.at/shim', 'array.prototype.at',
```
Comment (ljharb) [inline src/plugins/js-eval/jsEvalPlugin.js:67]:
```suggestion
      '-r', 'string.prototype.at/auto', 'array.prototype.at/auto',
```

File: src/plugins/js-eval/jsEvalPlugin.js [hunk 3]
```diff
@@ -64,6 +64,7 @@ module.exports = async function jsEvalPlugin({ mentionUser, respond, message })
       '--experimental-vm-modules', // used by m>
       '--experimental-modules',
       '--no-warnings',
+      '-r', 'string.prototype.at/shim', 'array.prototype.at/shim',
```
Comment (ljharb) [inline src/plugins/js-eval/jsEvalPlugin.js:67]:
```suggestion
      '-r', 'string.prototype.at/auto', 'array.prototype.at/auto',
```

The shim entry point does nothing, it’s just a function.

File: src/plugins/js-eval/jsEvalPlugin.js [hunk 4]
```diff
@@ -64,6 +64,7 @@ module.exports = async function jsEvalPlugin({ mentionUser, respond, message })
       '--experimental-vm-modules', // used by m>
       '--experimental-modules',
       '--no-warnings',
+      '-r', 'string.prototype.at/auto', 'array.prototype.at/auto',
```
Comment (brigand) [inline src/plugins/js-eval/jsEvalPlugin.js:67]:
```suggestion
      '-r', 'string.prototype.at/auto', '-r', 'array.prototype.at/auto',
```

**Comment by brigand** [comment]
I agree, this should only be enabled for `b>`. I use `n>` to check if node supports something.

**Comment by caub** [comment]
yes, will do, but I don't understand why the CI fails though

**Comment by brigand** [comment]
It's failing because you only have one `-r`. You need one before each thing you want to require.

**Comment by ljharb** [comment]
Do we also want to add https://npmjs.com/object.hasown, also stage 3?

**Comment by caub** [comment]
Sure, but first we need to figure why the CI fails, I don't have docker installed locally anymore so I can't test easily for now

By the way, I'm not sure to understand the added value of .at https://replit.com/@caub/Stringprototypeat#index.js I thought it would return a whole unicode char, unlike charAt

**Comment by ljharb** [comment]
No, it’s just using code units.

Code points aren’t useful anyways; you want grapheme clusters. 💩 is 1 code point but 🏳️‍🌈 is like 4.

**Comment by brigand** [comment]
Maybe an issue with the package [using "exports"](https://unpkg.com/browse/string.prototype.at@1.0.1/package.json) and `-r`? I was able to get the tests passing locally by loading them with `require()`. Will see if CI agrees.

**Comment by caub** [comment]
beautiful @brigand :+1:

**Pull request merged**